### PR TITLE
Add missing ilab model list help screen description

### PR DESCRIPTION
**Pull request by alinaryan**
**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by alinaryan** [comment]
New output:

```
Usage: ilab model list [OPTIONS]

  Lists models

Options:
  --model-dirs TEXT   Base directories where models are stored.  [default:
                      /home/alina/.cache/instructlab/models]
  --list-checkpoints  List the trained model checkpoints.
  --help              Show this message and exit.
```

File: src/instructlab/model/list.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 )
 @click.option(
     "--list-checkpoints",
+    help="List the trained model checkpoints.",
```
Comment (RobotSail) [inline src/instructlab/model/list.py:32]: You should probably reword this to "Also include trained model checkpoints" or something.

**Review by RobotSail — APPROVED** [review]
LGTM, just please change the wording a little bit - the reason is that the current wording makes it seem like it's listing the trained checkpoints _instead_ of the existing models.

File: src/instructlab/model/list.py (hunk 1 above)
Comment (alinaryan) [reply src/instructlab/model/list.py:32]: So it's listing both trained model checkpoints and existing models?

Comment (RobotSail) [reply src/instructlab/model/list.py:32]: Yeah it's not a binary flag, it's mainly a fallback case we check if the given path isn't a safetensors directory or gguf model:

https://github.com/instructlab/instructlab/blob/f0590b710689d0ae0051e744b47f2517b563a828/src/instructlab/model/list.py#L97

Comment (alinaryan) [reply src/instructlab/model/list.py]: How is this?

Comment (RobotSail) [reply src/instructlab/model/list.py]: I think the way you've written it is good for a command description, but we'll probably want to change it to "include the trained checkpoints in the output", since we're extending the existing functionality.

**Comment by danmcp** [comment]
@mergifyio rebase

**Review by booxter — APPROVED** [review]

File: src/instructlab/model/list.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
 )
 @click.option(
     "--list-checkpoints",
+    help="List the trained model checkpoints and all existing models.",
```
Comment (booxter) [inline src/instructlab/model/list.py:32]: since it's a description of the effect of this particular option, perhaps you could say smth like "Also list checkpoints [in addition to existing models]."

Comment (alinaryan) [reply src/instructlab/model/list.py]: Changed it to this

**Review by RobotSail — APPROVED** [review]
LGTM
